### ISSUE TITLE
[Bug]: Error: reading RAM Resource Share (arn:aws:/ram/:eu-west-2:xxxxxx:resource-share/"xxxxxx): couldn't find resource

### ISSUE THREAD

**Issue by Viktork25** [hashicorp/terraform-provider-aws#33701]
### Terraform Core Version

1.1.0

### AWS Provider Version

5.14.0

### Affected Resource(s)

Error: reading RAM Resource Share (arn:aws:/ram/:eu-west-xxxxxxx:resource-share/xxxxx): couldn't find resource

### Expected Behavior

We are trying to connect our transit gateway with a share transit gateway which has auto acceptance in place.The ram share-resource should be working and not tainted as it turns up and have a full transit gateway attachment shared

### Actual Behavior

The ram share-resource fails and says that cannot be finded but ch checking the AWS Console, that share-resource is active and is already accepted.

### Relevant Error/Panic Output Snippet

```shell
Error: reading RAM Resource Share (arn:aws:/ram/:eu-west-2:xxxxxxxxxxxxxx:resource-share/"xxxxxxxxxxxxxxxxxxxxxxx): couldn't find resource
```

### Terraform Configuration Files

```shell
module.vpc_eu_west_2[0].aws_/ram/_resource_share_accepter.this[0] is tainted, so must be replaced

+/- resource “aws_ram_resource_share_accepter” “this” {

 ~ id = “arn:aws:ram:eu-west-2:xxxxxxxx:resource-share/xxxxxxxxxxxxx” -> (known after apply)

 ~ invitation_arn = “arn:aws:/ram/:eu-west-2:xxxxxxxxxxxx:resource-share-invitation/xxxxxxxxxxxxxxx” -> (known after apply)

 ~ receiver_account_id = “xxxxxxxx” -> (known after apply)

 ~ resources = [

 - “arn:aws:ec2:eu-west-2:xxxxxxxxxxxxxx:transit-gateway/tgw-000000000000”,

 ] -> (known after apply)

 ~ sender_account_id = “xxxxxxxxxxx” -> (known after apply)

 ~ share_id = “rs-xxxxxxxdddddddddddd” -> (known after apply)

 ~ share_name = “T-SHARE” -> (known after apply)

 ~ status = “ACTIVE” -> (known after apply)


 # (1 unchanged attribute hidden)
```

### Steps to Reproduce

1. New transit gateway ram-share
2. Auto accepted from share account
3. Ram share is active
4. Rerun the pipeline to have the attachment and create the transit gateway in the receiver account

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @Viktork25  👋 Thank you for taking the time to raise this! Are you able to provide a sample Terraform configuration that exhibits this behavior, as well as debug logging? There's often helpful context in there for for whoever winds up looking into this.

**Comment by avakil95** [comment]
Hello, we are experiencing this same issue. 

The odd thing is, when the resource is attempted to get created, it does indeed accept the resource share invitation but fails to create with the error: 
Error: reading RAM Resource Share (arn:aws-us-gov:ram:us-gov-west-1:XXXXXXXXX:resource-share/XXXXX-XXXXXXX): couldn't find resource

When I run an apply on the resource again, terraform thinks the resource is tained and wants to replace it. If I apply the resource, the same error occurs and it never thinks the resource is fully created. 

I'm on Terraform v1.5.7 and AWS provider version 5.13.0, but I've also tried AWS provider version 5.32.1 and the same thing happens.

**Comment by Ilhicas** [comment]
Having the exact same error on version 5.16 and terraform 1.5.7
There are several references https://github.com/hashicorp/terraform-provider-aws/issues/18332 that state the same issue which was fixed in 3.38
